docs/next/components/mdx/RenderedDAG.tsx

```diff
@@ -0,0 +1,381 @@
+    };
+  }
+
+  getOffsetXY(e: MouseEvent | React.MouseEvent): Point | null {
+    const el = this.element.current;
+    if (!el) {
+      return null;
+    }
+    const ownerRect = el.getBoundingClientRect();
+    return {x: e.clientX - ownerRect.left, y: e.clientY - ownerRect.top};
+  }
+
+  public shiftXY(dx: number, dy: number) {
+    const {x, y, scale} = this.state;
+    this.setState({x: x + dx, y: y + dy, scale});
+  }
+
+  public adjustZoomRelativeToScreenPoint(nextScale: number, point: Point) {
+    const centerSVGCoord = this.screenToSVGCoords(point);
+    const {scale} = this.state;
+    let {x, y} = this.state;
+    x = x + (centerSVGCoord.x * scale - centerSVGCoord.x * nextScale);
+    y = y + (centerSVGCoord.y * scale - centerSVGCoord.y * nextScale);
+    this.setState({x, y, scale: nextScale});
+  }
+
+  public smoothZoom(to: {x: number; y: number; scale: number}) {
+    const from = {scale: this.state.scale, x: this.state.x, y: this.state.y};
+
+    if (this._animation) {
+      this._animation.cancel();
+    }
+
+    this._animation = animate(from, to, {
+      step: (v: any) => {
+        this.setState({
+          x: v.x,
+          y: v.y,
+          scale: v.scale,
+        });
+      },
+      done: () => {
+        this.setState(to);
+        this._animation = null;
+      },
+    });
+  }
+
+  public getMaxZoom() {
+    return this.props.maxZoom;
+  }
+
+  onDoubleClick = (event: React.MouseEvent<HTMLDivElement>) => {
+    // Don't allow double-click events on the zoom slider to trigger this.
+    if (event.target instanceof HTMLElement && event.target.closest('#zoom-slider-container')) {
+      return;
+    }
+    this.autocenter(true);
+    event.stopPropagation();
+  };
+
+  render() {
+    const {x, y, scale} = this.state;
+    const dotsize = Math.max(14, 30 * scale);
+
+    return (
+      <div
+        tabIndex={-1}
+        ref={this.element}
+        style={Object.assign({}, SVGViewportStyles, {
+          backgroundPosition: `${x}px ${y}px`,
+          backgroundSize: `${dotsize}px`,
+        })}
+        onMouseDown={(e) => this.onMouseDown(this, e)}
+        onDoubleClick={this.onDoubleClick}
+        onFocus={() => {
+          document.addEventListener('wheel', this.onWheel, {passive: false});
+          this.props.onFocus();
+        }}
+        onBlur={() => {
+          document.removeEventListener('wheel', this.onWheel);
+          this.props.onBlur();
+        }}
+      >
+        <div
+          dangerouslySetInnerHTML={{__html: this.props.content}}
+          style={{
+            transformOrigin: `top left`,
+            transform: `matrix(${scale}, 0, 0, ${scale}, ${x}, ${y})`,
+            pointerEvents: 'none',
+          }}
+        ></div>
+      </div>
+    );
+  }
+}
+
+/*
+  BG: Not using styled-components here because I need a `ref` to an actual DOM element.
+  Styled-component with a ref returns a React component we need to findDOMNode to use.
+  */
```

Comment:
Is styled-components used in the docs site? I don't see it in package.json.

Reply:
ah no, this is a casualty of copypasta